dashboard/src/apollo/httpLink.js

```diff
@@ -1,10 +1,34 @@
 import { BatchHttpLink as HttpLink } from "apollo-link-batch-http";
 import doFetch from "/utils/fetch";
+import gql from "graphql-tag";
 
-const httpLink = () =>
+const mutation = gql`
+  mutation SetLocalNetworkOfflineMutation($offline: Boolean!) {
+    setLocalNetworkOffline(offline: $offline) @client
+  }
+`;
+
+const httpLink = ({ getClient }) =>
   new HttpLink({
     uri: "/graphql",
-    fetch: doFetch,
+    fetch: (url, init) =>
+      doFetch(url, init).then(
+        response => {
+          getClient().mutate({
+            mutation,
+            variables: { offline: true },
+          });
+          return response;
+        },
+        error => {
+          getClient().mutate({
```

Comment:
There is a check here that we are missing.

We need to test that the error is an instance of `FailedError` in order to isolate connection errors from other http errors.

(See https://github.com/sensu/sensu-go/blob/master/dashboard/src/utils/fetch.js for the heuristic)

Reply:
Updated!